features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SessionReplayConfiguration.kt

```diff
@@ -63,8 +66,35 @@ data class SessionReplayConfiguration internal constructor(
          * @see SessionReplayPrivacy.MASK
          * @see SessionReplayPrivacy.MASK_USER_INPUT
          */
+        @Deprecated(
+            message = "This method is deprecated and will be removed in future versions. " +
+                "Use the new fine grained masking apis instead: " +
+                "setImagePrivacy, setTouchPrivacy, setTextAndInputPrivacy."
+        )
         fun setPrivacy(privacy: SessionReplayPrivacy): Builder {
-            this.privacy = privacy
+            // if fgm levels have already been explicitly set then ignore legacy privacy.
+            if (fineGrainedMaskingModified) return this
```

Comment:
minor: maybe "fineGrainedMaskingSet" or "fineGrainedMaskingUsed" as the name instead? When I saw it in the declaration, it makes me wonder a while what's for.

Reply:
sgtm - I'll also add an explanatory comment at the declaration